src/monattack.cpp

```diff
@@ -247,8 +247,11 @@ bool mattack::eat_crop( monster *z )
 {
     for( const auto &p : g->m.points_in_radius( z->pos(), 1 ) ) {
         if( g->m.has_flag( "PLANT", p ) && one_in( 4 ) ) {
-            g->m.ter_set( p, t_dirt );
-            g->m.furn_set( p, f_null );
+            if( g->m.furn( p ).obj().plant_base != "" ) {
```

Comment:
'!if( g->m.furn( p ).obj().plant_base.empty()'

Reply:
Yes, but `!` jumped out of the condition expression.
```cpp
if( !g->m.furn( p ).obj().plant_base.empty() ) {
```

Reply:
Yeah, you're right.